synthesis/codon/analyzer_test.go

```diff
@@ -0,0 +1,57 @@
+package codon
+
+import (
+	"testing"
+
+	"github.com/TimothyStiles/poly/io/genbank"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestComputeStats(t *testing.T) {
+	t.Parallel()
+
+	sortStrings := cmpopts.SortSlices((func(a, b string) bool { return a < b }))
+
+	tests := []struct {
+		name string
+
+		testDataPath string
+		wantStats    *Stats
+
+		wantErr error
+	}{
+		{
+			name: "pUC19 plasmid",
+
+			testDataPath: "../../data/puc19.gbk",
+
+			wantStats: &Stats{
+				StartCodonCount: map[string]int{"atg": 2},
+			},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		var tt = tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			sequence, err := genbank.Read(tt.testDataPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			analyzer := &Analyzer{}
+
+			stats, err := analyzer.ComputeStats(sequence)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
IMO this shows why the `Analyzer` struct is unnecessary. You're instantiating an empty struct and immediately calling one of its methods, then never use the struct again.

Reply:
the test is meant to only test `ComputeStats`, so which other method would I be calling? The point of the single method struct is to make mocking/interfacing easier for consumers.

Reply:
> the test is meant to only test ComputeStats, so which other method would I be calling?

Oop yeah, nevermind on this.

> The point of the single method struct is to make mocking/interfacing easier for consumers.

I still don't see the the use case for this. I'll leave a more detailed response in the main PR thread.